### DISCUSSION TITLE
how to run specific tests?

### DISCUSSION THREAD

**Discussion by jwaldmann** [tidalcycles/Tidal#1130] [General]
Is there a way to select tests for execution? I was thinking of
```
stack  --resolver=nightly test --test-arguments='*VF*' --file-watch
```
(https://hackage.haskell.org/package/tasty allows patterns, microspec probably does not)

I have this work-around now
```
stack  --resolver=nightly exec -- ghcid -c ghci -W -T"microspec Sound.Tidal.ParamsTest.run" -- -isrc:test  Sound.Tidal.ParamsTest
```
which seems good enough. 

Put something like this in CONTRIBUTING?

**Comment by sss-create** [comment]
something for cabal would be nice. On the other side, the number of tests is not that big, to slow down the testing process

**Comment by yaxu** [comment]
AFAICT, tidal is the only project on hackage to use microspec, and it hasn't been updated in over 6 years.. If we want missing features, we should probably move to a more standard test framework.
